main.js

```diff
@@ -160,4 +160,152 @@ const inventory = [
     originalStock: 10,
     sold: 8,
   },
-];
\ No newline at end of file
+];
+
+// ---------------- OPDRACHT 1: ARRAY METHODEN ----------------------
+
+// 1A
+const tvNames = inventory.map((tv) => {
+  return tv.name;
+});
+console.log(tvNames);
+
+//1B
+const tvSoldOut = inventory.filter((tv) => {
+  return tv.originalStock - tv.sold === 0;
+});
+console.log(tvSoldOut);
+
+//1C
+const tvAmbiLight = inventory.filter((tv) => {
+  return tv.options.ambiLight === true;
+});
+console.log(tvAmbiLight);
+
+//1D
+const tvHighestPriceToLowest = inventory.sort((a, b) => {
+  return a.price - b.price;
+});
+console.log(tvHighestPriceToLowest);
+
+
+
+// ---------------- OPDRACHT 2: ELEMENTEN IN DE DOM PLAATSEN ----------------------
+
+//2A
+let tvsSold = 0;
+for (let i = 0; i<inventory.length; i++) {
+  tvsSold += inventory[i].sold;
+}
+console.log(tvsSold);
+
+//2B
+tvsSoldHTML = document.getElementById("tvs-sold");
+tvsSoldHTML.textContent = tvsSold;
+
+//2C
+let tvsBought = 0;
+for (let i = 0; i<inventory.length; i++) {
+  tvsBought += inventory[i].originalStock;
+}
+console.log(tvsBought);
+
+//2D
+tvsBoughtHTML = document.getElementById("tvs-bought");
+tvsBoughtHTML.textContent = tvsBought;
+
+//2E
+let tvsToBeSold = tvsBought-tvsSold;
+tvsSoldHTML = document.getElementById("tvs-to-be-sold");
+tvsSoldHTML.textContent = tvsToBeSold;
+
+
+// ---------------- OPDRACHT 3 ARRAY METHODEN EN FUNCTIES ----------------------
+
+//3A
+// const tvBrands = inventory.map((tv) => {
+//   return tv.brand + " ";
+// });
+// tvBrandsHTML = document.getElementById("tv-brands");
+// tvBrandsHTML.textContent = tvBrands;
+// console.log(tvBrands);
+
+//3B
+function showTvBrands () {
+  const tvBrands = inventory.map((tv) => {
+    return tv.brand + " ";
+  });
+  tvBrandsHTML = document.getElementById("tv-brands");
+  tvBrandsHTML.textContent = tvBrands;
+  console.log(tvBrands);
+}
+
+showTvBrands();
+
+
+
+// ---------------- OPDRACHT 4 FUNCTIONS----------------------
+
+//4A
+let tvInfo= "";
+function showTvInfo (tv) {
+ return tvInfo = `${tv.brand} ${tv.type} - ${tv.name}`;
+}
+console.log(showTvInfo(inventory[4]));
+
+// 4B
+function showTvPrice (tv) {
+  let tvPrice = "";
+  return tvPrice = `€${tv.price},-
+  `
+}
+console.log(showTvPrice(inventory[4]));
+
+//4C
+function availableScreenSizes (tv) {
+  let tvScreenSize = "";
+  for (let i = 0; i <tv.availableSizes.length; i++) {
+    if (i === 0) {
+      tvScreenSize += `${tv.availableSizes[i]} inch (${tv.availableSizes[i] * 2.5} cm)`;
+    }
+    else { tvScreenSize += `| ${tv.availableSizes[i]} inch (${tv.availableSizes[i] * 2.5} cm)`;
+    }
```

Comment:
Om code duplicatie te reduceren kun je het op zo'n manier aanpakken
```suggestion
    tvScreenSize += `${tv.availableSizes[i]} inch (${tv.availableSizes[i] * 2.5} cm)`;
    
    if (i > 0) {
        tvScreenSize += " | ";
    }
```